src/azure/cosmos/partition_key.rs

```diff
@@ -59,3 +51,25 @@ impl<'a> IntoIterator for PartitionKey<'a> {
         }
     }
 }
+
```

Comment:
This is weird. I've run into this String &String &str dance myself but I've been unable to find a solution either :face_with_head_bandage: